lib/cib/cib_attrs.c

```diff
@@ -120,48 +106,45 @@ find_attr(cib_t *cib, const char *section, const char *node_uuid,
             node_type = XML_CIB_TAG_STATE;
             set_type = XML_TAG_TRANSIENT_NODEATTRS;
         }
-        attr_snprintf(xpath_string, offset, XPATH_MAX, "//%s[@id='%s']", node_type,
-                      node_uuid);
+        g_string_append_printf(xpath, "//%s[@" XML_ATTR_ID "='%s']", node_type,
+                               node_uuid);
     }
 
     if (set_name) {
-        attr_snprintf(xpath_string, offset, XPATH_MAX, "//%s[@id='%.128s']", set_type,
```

Comment:
You're getting rid of the `.128` here and several other places.  Is that alright?

Reply:
I presume so, since the goal is to build an XPath string to match something. My expectation is that the current version would simply fail to match its intended target, if our string doesn't fit into 128 characters. The `.128` was added in 4d978f5, and no rationale was given beyond ensuring that the string fits within the buffer.